src/main/kotlin/uk/gov/justice/digital/hmpps/nomisvisitsmappingservice/coreperson/CorePersonMappingResource.kt

```diff
@@ -0,0 +1,106 @@
+package uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.coreperson
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import jakarta.validation.Valid
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.dao.DuplicateKeyException
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.config.DuplicateMappingErrorResponse
+import uk.gov.justice.digital.hmpps.nomisvisitsmappingservice.config.DuplicateMappingException
+import uk.gov.justice.hmpps.kotlin.common.ErrorResponse
+
+@RestController
+@Validated
+@PreAuthorize("hasRole('NOMIS_CORE_PERSON')")
+@RequestMapping("/mapping/core-person", produces = [MediaType.APPLICATION_JSON_VALUE])
+class CorePersonMappingResource(private val service: CorePersonService) {
+  private companion object {
+    val log: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  @PostMapping("/migrate")
```

Comment:
Currently a post/create endpoint but could possibly be made into upsert to allow for subsequence migrations.